com.unity.render-pipelines.universal/Samples~/TeachingSamples/SharedAssets/Scripts/FPSController.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Rigidbody))]
+public class FPSController : MonoBehaviour
+{
+    public bool moveWithMouse = true;
+    public float mouseSensitivity = 1;
+    public float buttonSensitivity = 1;
+    public float walkSpeed = 1;
+
+    private Rigidbody rb;
+    private byte movementFlags;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        if (moveWithMouse)
+        {
+            Cursor.visible = false;
+            Cursor.lockState = CursorLockMode.Locked;
+        }
+        
+        rb = GetComponent<Rigidbody>();
+
+        movementFlags = 0;
+    }
+
+    private void Update()
+    {
+        if (moveWithMouse)
+        {
+            //Rotate based on mouse input
+            float vert = Input.GetAxis("Mouse X") * Time.deltaTime * mouseSensitivity * 100;
+            float hori = Input.GetAxis("Mouse Y") * Time.deltaTime * mouseSensitivity * 100;
```

Comment:
This will throw errors if the user enabled the new input system package. should we support this?

https://docs.unity3d.com/Manual/com.unity.inputsystem.html

Reply:
The legacy input code has been wrapped in an ifdef